### PR TITLE
Customize status plot for HG PCL alignment

### PR DESCRIPTION
**Pull request by dmeuser**
#### PR description:
In this PR the status plot used for the HG PCL alignment is customized to properly show the alignment update decision based on the fraction threshold logic, which is currently used. The "old" status plot is now only used for the LG PCL alignment.
In detail the following changes have been made:

- Add a map to `MillePedeFileReader` to store the threshold decision per structure and direction. Introduce a method to get the map.
- In `MillePedeDQMModule` different versions status plot depending on the alignment granularity (LG or HG) are produced.

An example of the customized status plot is shown below. Here, the fraction threshold is exceeded for three different detector-direction combinations and the alignment would be updated. The DQM GUI render plugin will be updated to cope with the "new" status plot.

image

#### PR validation:
The PR can be validated using `runTheMatrix.py -l 1001.2 --ibeos`.

@mmusich, @connorpa, @antoniovagnerini, @consuegs

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * workflow = 1001.2

**Comment by tvami** [comment]
type new-feature,trk

**Comment by mmusich** [comment]
@tvami I would argue it's a bug-fix rather than a new feature.

File: Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeDQMModule.cc [hunk 1]
```diff
@@ -188,6 +192,26 @@ void MillePedeDQMModule ::fillStatusHisto(MonitorElement* statusHisto) {
   histo_status->GetXaxis()->SetBinLabel(6, "above significance");
 }
 
+void MillePedeDQMModule ::fillStatusHistoHG(MonitorElement* statusHisto) {
+  TH2F* histo_status = statusHisto->getTH2F();
+  auto theResults = mpReader_->getResultsHG();
```
Comment (ChrisMisan) [inline Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeDQMModule.cc:197]:
```suggestion
  auto& theResults = mpReader_->getResultsHG();
```

File: Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h [hunk 2]
```diff
@@ -108,6 +108,8 @@ class MillePedeFileReader {
     return mpPCLresults(updateDB_, vetoUpdateDB_, Nrec_, exitCode_, exitMessage_, updateBits_);
   }
 
+  const std::map<std::string, std::array<bool, 6>> getResultsHG() const { return fractionExceeded_; }
```
Comment (ChrisMisan) [inline Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h:111]:
```suggestion
  const std::map<std::string, std::array<bool, 6>>& getResultsHG() const { return fractionExceeded_; }
```
Wouldn't it be better from the memory performance perspective to have it passed by reference if it's const either way?

File: Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeDQMModule.cc (hunk 1 above)
Comment (dmeuser) [reply Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeDQMModule.cc]: Done

File: Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h (hunk 2 above)
Comment (dmeuser) [reply Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h]: Done

**Comment by mmusich** [comment]
please test

**Comment by tvami** [comment]
@cmsbuild , please test
 * the ongoing test will have an input error

**Comment by ChrisMisan** [comment]
+alca

- differences only in msgLogger and MTD
- 1001.2 validation passes
- static checks outside of this PR scope

**Comment by qliphy** [comment]
+1

**Pull request merged**